documentation/collectionruleexamples.md

```diff
@@ -0,0 +1,261 @@
+# Collection Rule Examples
+
+The following examples provide sample scenarios for using a collection rule. These templates can be copied directly into your configuration file with minimal adjustments to work with your application (for more information on configuring an egress provider, see [egress providers](./configuration.md#egress-configuration)), or they can be adjusted for your specific use-case. [Learn more about configuring collection rules](collectionrules.md).
+
+## Collect Trace - Startup (`Startup` Trigger)
+
+### JSON
+
+```json
+{
+  "AssemblyLoadTraceOnStartup": {
+    "Trigger": {
+      "Type": "Startup"
+    },
+    "Actions": [
+      {
+        "Type": "CollectTrace",
+        "Settings": {
+          "Providers": [{
+              "Name": "Microsoft-Windows-DotNETRuntime",
+              "EventLevel": "Informational",
+              "Keywords": "0x8"
+          }],
+          "Duration": "00:00:15",
+          "Egress": "artifacts"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Explanation
+
+This rule, named "AssemblyLoadTraceOnStartup", will trigger on a process's startup. When the rule is triggered, a trace will be collected for 15 seconds and egressed to the specified `Egress` provider (in this case, `artifacts` has been configured to save the trace to the local filesystem). The trace will capture events from an event provider named `Microsoft-Windows-DotNETRuntime`, and will collect events at or above the `Informational` level using the keyword `0x8` (`LoaderKeyword`). The trace will request rundown by default, and the `BufferSizeInMB` has the default value of 256 MB.
+
+## Collect GCDump - Heap Size (`EventCounter` Trigger)
+
+### JSON
+
+```json
+{
+  "LargeGCHeapSize": {
+    "Trigger": {
+      "Type": "EventCounter",
+      "Settings": {
+        "ProviderName": "System.Runtime",
+        "CounterName": "gc-heap-size",
+        "GreaterThan": 1
+      }
+    },
+    "Actions": [
+      {
+        "Type": "CollectGCDump",
+        "Settings": {
+          "Egress": "artifacts"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Explanation
+
+This rule, named "LargeGCHeapSize", will trigger when the GC Heap Size exceeds 1 MB within the default sliding window duration (1 minute). If the rule is triggered, a GCDump will be collected and egressed to the specified `Egress` provider (in this case, `artifacts` has been configured to save the GCDump to the local filesystem). There is a default `ActionCount` limit stating that this rule may only be triggered 5 times.
```

Comment:
Not something that needs to be added here but https://github.com/dotnet/dotnet-monitor/issues/1126 does affect gcdumps right now